src/renderer/components/+namespaces/namespace-select-filter.tsx

```diff
@@ -49,14 +61,22 @@ const Placeholder = observer((props: PlaceholderProps<any, boolean>) => {
   };
 
   return (
-    <components.Placeholder {...props}>
+    <components.Placeholder {...props} data-testid="namespace-select-filter-placeholder">
       {getPlaceholder()}
     </components.Placeholder>
   );
 });
 
+export interface NamespaceSelectFilterProps extends NamespaceSelectProps {
+  maxItems?: number;
+}
+
+interface NamespaceSelectFilterDependencies {
+  namespaceStore: NamespaceStore,
+}
+
 @observer
-export class NamespaceSelectFilter extends React.Component<SelectProps> {
+export class NonInjectedNamespaceSelectFilter extends React.Component<NamespaceSelectFilterProps & NamespaceSelectFilterDependencies> {
```

Comment:
```suggestion
export class NamespaceSelectFilter extends React.Component<NamespaceSelectFilterProps> {
// ...

// export injected version as default
export default withInjectables(...);

// or export by explicit name
export const NamespaceSelectFilterInjected = withInjectables(...);

```

Reply:
This currently follows the style that we have used elsewhere for injectable components.

Personally I like spitting them apart like this as it means that there is both a smaller diff and we can talk about the injected vs the injectable version of a component by name.